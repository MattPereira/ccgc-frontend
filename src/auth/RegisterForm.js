import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { Card, Form, Alert } from "react-bootstrap";

/** Register form.
 *
 * Shows form and manages update to state on changes.
 * On submission:
 * - calls register function prop
 * - redirects to "/" route
 *
 * Routes -> RegisterForm
 * Routed to "/register"
 */

const RegisterForm = ({ register }) => {
  const navigate = useNavigate();
  const [formData, setFormData] = useState({
    email: "",
    password: "",
    firstName: "",
    lastName: "",
  });

  const [formErrors, setFormErrors] = useState([]);

  console.debug(
    "RegisterForm",
    "register=",
    typeof register,
    "formData=",
    formData,
    "formErrors=",
    formErrors
  );

  //update state of formData onChange of any form input field
  const handleChange = (e) => {
    const { name, value } = e.target;

    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    //register function passes form data up to parent App component
    let result = await register(formData);
    if (result.success) {
      navigate("/");
    } else {
      setFormErrors(result.errors);
    }
  };

  return (
    <div className="row justify-content-center">
      <div className="col-sm-8">
        <Card className="shadow mb-5">
          <Card.Title className="display-2 text-center bg-dark text-white py-2 mb-0">
            Register
          </Card.Title>
          <Card.Img
            src="https://images.unsplash.com/photo-1605144156683-5ebde77feed5?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1742&q=80"
            alt="Register card image"
          />
          <Card.Body className="px-5 py-5">
            <Form onSubmit={handleSubmit}>
              <Form.Group className="mb-3">
                <Form.Label htmlFor="email">Email</Form.Label>
                <input
                  className="form-control"
                  id="email"
                  name="email"
                  type="email"
                  value={formData.email}
                  onChange={handleChange}
                  autoComplete="email"
                  required
                ></input>
              </Form.Group>
              <Form.Group className="mb-3">
                <Form.Label htmlFor="password">Password</Form.Label>
                <input
                  className="form-control"
                  id="password"
                  name="password"
                  type="password"
                  value={formData.password}
                  onChange={handleChange}
                  autoComplete="current-password"
                  required
                ></input>
              </Form.Group>
              <Form.Group className="mb-3">
                <Form.Label htmlFor="firstName">First Name</Form.Label>
                <input
                  className="form-control"
                  id="firstName"
                  name="firstName"
                  type="text"
                  value={formData.firstName}
                  onChange={handleChange}
                  required
                ></input>
              </Form.Group>
              <Form.Group className="mb-3">
                <Form.Label htmlFor="lastName">Last Name</Form.Label>
                <input
                  className="form-control"
                  id="lastName"
                  name="lastName"
                  type="text"
                  value={formData.lastName}
                  onChange={handleChange}
                  required
                ></input>
              </Form.Group>

              {formErrors.length
                ? formErrors.map((err) => (
                    <Alert color="danger" key={err}>
                      {err}
                    </Alert>
                  ))
                : null}

              <div className="row justify-content-end">
                <div className="col-auto">
                  <button className="btn btn-primary btn-block px-4">
                    Register
                  </button>
                </div>
              </div>
            </Form>
          </Card.Body>
        </Card>
      </div>
    </div>
  );
};

export default RegisterForm;
