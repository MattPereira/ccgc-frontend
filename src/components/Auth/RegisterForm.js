import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { Form, Alert } from "react-bootstrap";
import {
  Container,
  Button,
  Typography,
  Paper,
  Box,
  TextField,
  Grid,
} from "@mui/material";
import { styled } from "@mui/material/styles";
import ccgclogo from "../../assets/ccgc_logo.png";

/** Register form.
 *
 * Shows form and manages update to state on changes.
 * On submission:
 * - calls register function prop
 * - redirects to "/" route
 *
 * Routes -> RegisterForm
 * Routed to "/register"
 */

const StyledPaper = styled(Paper)(({ theme }) => ({
  borderRadius: "30px",
  backgroundColor: "#eeeeee",
}));

const StyledCardImage = styled(Box)(({ theme }) => ({
  width: "50%",
  borderRadius: "30px",
}));

export default function RegisterForm({ register }) {
  const navigate = useNavigate();
  const [formData, setFormData] = useState({
    email: "",
    password: "",
    firstName: "",
    lastName: "",
  });

  const [formErrors, setFormErrors] = useState([]);

  console.debug(
    "RegisterForm",
    "register=",
    typeof register,
    "formData=",
    formData,
    "formErrors=",
    formErrors
  );

  //update state of formData onChange of any form input field
  const handleChange = (e) => {
    const { name, value } = e.target;

    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    //register function passes form data up to parent App component
    let result = await register(formData);
    if (result.success) {
      navigate("/");
    } else {
      setFormErrors(result.errors);
    }
  };

  return (
    <Container className="py-5">
      <Grid container justifyContent="center">
        <Grid item xs={12} md={6}>
          <StyledPaper elevation={5}>
            <Box
              sx={{
                backgroundColor: "black",
                borderRadius: "30px",
                textAlign: "center",
                py: 3,
              }}
            >
              <StyledCardImage component="img" src={ccgclogo} />
            </Box>
            <Box sx={{ p: 3 }}>
              <Form onSubmit={handleSubmit}>
                <Typography variant="h1" sx={{ mb: 5 }}>
                  Register
                </Typography>
                <Box sx={{ mb: 3 }}>
                  <TextField
                    id="email"
                    name="email"
                    label="Email"
                    type="text"
                    variant="outlined"
                    onChange={handleChange}
                    required
                    autoComplete="email"
                    sx={{ width: "100%" }}
                  />
                </Box>
                <Box sx={{ mb: 3 }}>
                  <TextField
                    id="password"
                    name="password"
                    type="password"
                    label="Password"
                    variant="outlined"
                    onChange={handleChange}
                    required
                    autoComplete="email"
                    sx={{ width: "100%" }}
                  />
                </Box>
                <Box sx={{ mb: 3 }}>
                  <TextField
                    id="firstName"
                    name="firstName"
                    type="text"
                    label="First Name"
                    variant="outlined"
                    onChange={handleChange}
                    required
                    sx={{ width: "100%" }}
                  />
                </Box>
                <Box sx={{ mb: 3 }}>
                  <TextField
                    id="lastName"
                    name="lastName"
                    type="text"
                    label="Last Name"
                    variant="outlined"
                    onChange={handleChange}
                    required
                    sx={{ width: "100%" }}
                  />
                </Box>

                {formErrors.length
                  ? formErrors.map((err) => (
                      <Alert variant="danger" key={err}>
                        {err}
                      </Alert>
                    ))
                  : null}

                <Box sx={{ textAlign: "end" }}>
                  <Button
                    variant="contained"
                    type="submit"
                    sx={{ borderRadius: "30px", px: 3, py: 1 }}
                  >
                    Submit
                  </Button>
                </Box>
              </Form>
            </Box>
          </StyledPaper>
        </Grid>
      </Grid>
    </Container>
  );
}
