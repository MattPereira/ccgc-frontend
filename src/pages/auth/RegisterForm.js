import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { Form, Alert } from "react-bootstrap";
import {
  Container,
  Button,
  Typography,
  Paper,
  Box,
  TextField,
  Grid,
} from "@mui/material";
import { styled } from "@mui/material/styles";
import { Link } from "react-router-dom";

import SiteHero from "../../components/SiteHero";

/** Register form.
 *
 * Shows form and manages update to state on changes.
 * On submission:
 * - calls register function prop
 * - redirects to "/" route
 *
 * Routes -> RegisterForm
 * Routed to "/register"
 */

const StyledPaper = styled(Paper)(({ theme }) => ({
  borderRadius: "30px",
  backgroundColor: "#eeeeee",
}));

const StyledTextField = styled(TextField)({
  backgroundColor: "white",
  width: "100%",
});

export default function RegisterForm({ register }) {
  const navigate = useNavigate();
  const [formData, setFormData] = useState({
    email: "",
    password: "",
    firstName: "",
    lastName: "",
  });

  const [formErrors, setFormErrors] = useState([]);

  console.debug(
    "RegisterForm",
    "register=",
    typeof register,
    "formData=",
    formData,
    "formErrors=",
    formErrors
  );

  //update state of formData onChange of any form input field
  const handleChange = (e) => {
    const { name, value } = e.target;

    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    //register function passes form data up to parent App component
    let result = await register(formData);
    if (result.success) {
      navigate("/");
    } else {
      setFormErrors(result.errors);
    }
  };

  return (
    <Box>
      <SiteHero />
      <Container sx={{ py: 5 }}>
        <Grid container justifyContent="center">
          <Grid item xs={12} md={6}>
            <StyledPaper elevation={0}>
              <Box sx={{ p: 3 }}>
                <Form onSubmit={handleSubmit}>
                  <Typography variant="h1" sx={{ mb: 5 }}>
                    Register
                  </Typography>
                  <Box sx={{ textAlign: "center", my: 5 }}>
                    <Typography variant="p">
                      Already have an account?{" "}
                      <Link to="/login">Login here</Link>
                    </Typography>
                  </Box>
                  <Box sx={{ mb: 3 }}>
                    <StyledTextField
                      id="email"
                      name="email"
                      label="Email"
                      type="text"
                      variant="outlined"
                      onChange={handleChange}
                      required
                      autoComplete="email"
                      sx={{ width: "100%", bgcolor: "white" }}
                    />
                  </Box>
                  <Box sx={{ mb: 3 }}>
                    <StyledTextField
                      id="password"
                      name="password"
                      type="password"
                      label="Password"
                      variant="outlined"
                      onChange={handleChange}
                      required
                      sx={{ width: "100%" }}
                    />
                  </Box>
                  <Box sx={{ mb: 3 }}>
                    <StyledTextField
                      id="firstName"
                      name="firstName"
                      type="text"
                      label="First Name"
                      variant="outlined"
                      onChange={handleChange}
                      required
                      sx={{ width: "100%" }}
                    />
                  </Box>
                  <Box sx={{ mb: 3 }}>
                    <StyledTextField
                      id="lastName"
                      name="lastName"
                      type="text"
                      label="Last Name"
                      variant="outlined"
                      onChange={handleChange}
                      required
                      sx={{ width: "100%" }}
                    />
                  </Box>

                  {formErrors.length
                    ? formErrors.map((err) => (
                        <Alert variant="danger" key={err}>
                          {err}
                        </Alert>
                      ))
                    : null}

                  <Box sx={{ textAlign: "end" }}>
                    <Button
                      variant="contained"
                      type="submit"
                      sx={{ borderRadius: "30px", px: 3, py: 1 }}
                    >
                      Submit
                    </Button>
                  </Box>
                </Form>
              </Box>
            </StyledPaper>
          </Grid>
        </Grid>
      </Container>
    </Box>
  );
}
