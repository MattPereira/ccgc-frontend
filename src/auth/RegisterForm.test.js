import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import RegisterForm from "./RegisterForm";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <RegisterForm />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <RegisterForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays form inputs and register button", function () {
  render(
    <MemoryRouter>
      <RegisterForm />
    </MemoryRouter>
  );

  const emailInput = screen.getByLabelText("Email");
  expect(emailInput).toBeInTheDocument();

  const passwordInput = screen.getByLabelText("Password");
  expect(passwordInput).toBeInTheDocument();

  const firstNameInput = screen.getByLabelText("First Name");
  expect(firstNameInput).toBeInTheDocument();

  const lastNameInput = screen.getByLabelText("Last Name");
  expect(lastNameInput).toBeInTheDocument();

  //only one button in the component
  const registerBtn = screen.getByRole("button");
  expect(registerBtn).toBeInTheDocument();
});
