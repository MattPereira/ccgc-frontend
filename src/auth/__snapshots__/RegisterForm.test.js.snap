// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row justify-content-center"
  >
    <div
      class="col-sm-8"
    >
      <div
        class="shadow mb-5 card"
      >
        <div
          class="display-2 text-center bg-dark text-white py-2 mb-0 card-title h5"
        >
          Register
        </div>
        <img
          alt="Register card image"
          class="card-img"
          src="https://images.unsplash.com/photo-1605144156683-5ebde77feed5?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1742&q=80"
        />
        <div
          class="px-5 py-5 card-body"
        >
          <form
            class=""
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="email"
              >
                Email
              </label>
              <input
                autocomplete="email"
                class="form-control"
                id="email"
                name="email"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="current-password"
                class="form-control"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="firstName"
              >
                First Name
              </label>
              <input
                class="form-control"
                id="firstName"
                name="firstName"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="lastName"
              >
                Last Name
              </label>
              <input
                class="form-control"
                id="lastName"
                name="lastName"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="row justify-content-end"
            >
              <div
                class="col-auto"
              >
                <button
                  class="btn btn-primary btn-block px-4"
                >
                  Register
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
